### DISCUSSION TITLE
V1.1 support for rust gossiupsub

### DISCUSSION THREAD

**Discussion by kiransxt** [libp2p/rust-libp2p#2825] [General]
From the specs [page](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub#implementation-status) for gossipsub it seems rust v1.1 support is in progress. Would like know whether it is in active development and what is the current status?

**Comment by kiransxt** [comment]
I browsed through the code and saw parts of v1.1 specs implemented. As the specs implementation status page states its in active development, I am not very sure what parts of v1.1 is already supported and what is missing. 

@elenaf9 @mxinden @thomaseizinger

**Reply by thomaseizinger** [reply]
To my knowledge, gossipsub 1.1 is fully implemented and it is probably just the docs page that is outdated. @AgeManning can you provide more details here?

**Reply by mxinden** [reply]
I created https://github.com/libp2p/specs/pull/435.

**Reply by kiransxt** [reply]
Thanks a lot @thomaseizinger for confirming and @mxinden for the PR!!

**Comment by AgeManning** [comment]
Yep, kinda. 

Everything is fully implemented, except for peer-exchange during pruning, because at the time we didn't have signed peer records. As far as I'm aware, no one is really using this feature, so the drive to implement it is low. 

I will get there soon, just to say we fully support everything. 

See here: https://github.com/libp2p/rust-libp2p/issues/2398
